src/GraphQL/Federation/Extensions/FederationHelper.cs

```diff
@@ -0,0 +1,165 @@
+            inaccessibleDirective.Locations.Add(DirectiveLocation.ArgumentDefinition);
+            inaccessibleDirective.Locations.Add(DirectiveLocation.Scalar);
+            inaccessibleDirective.Locations.Add(DirectiveLocation.Enum);
+            inaccessibleDirective.Locations.Add(DirectiveLocation.EnumValue);
+            inaccessibleDirective.Locations.Add(DirectiveLocation.InputObject);
+            inaccessibleDirective.Locations.Add(DirectiveLocation.InputFieldDefinition);
+            schema.Directives.Register(inaccessibleDirective);
+        }
+
+        if (import.HasFlag(FederationDirectiveEnum.Override))
+        {
+            var overrideDirective = new Directive(OVERRIDE_DIRECTIVE)
+            {
+                Arguments = new(new QueryArgument<NonNullGraphType<StringGraphType>>() { Name = FROM_ARGUMENT }),
+            };
+            overrideDirective.Locations.Add(DirectiveLocation.FieldDefinition);
+            schema.Directives.Register(overrideDirective);
+        }
+
+        if (import.HasFlag(FederationDirectiveEnum.External))
+        {
+            var externalDirective = new Directive(EXTERNAL_DIRECTIVE);
+            externalDirective.Locations.Add(DirectiveLocation.FieldDefinition);
+            externalDirective.Locations.Add(DirectiveLocation.Object);
+            schema.Directives.Register(externalDirective);
+        }
+
+        if (import.HasFlag(FederationDirectiveEnum.Provides))
+        {
+            var providesDirective = new Directive(PROVIDES_DIRECTIVE)
+            {
+                Arguments = new(new QueryArgument<NonNullGraphType<FieldSetGraphType>>() { Name = FIELDS_ARGUMENT }),
+            };
+            providesDirective.Locations.Add(DirectiveLocation.FieldDefinition);
+            schema.Directives.Register(providesDirective);
+        }
+
+        if (import.HasFlag(FederationDirectiveEnum.Requires))
+        {
+            var requiresDirective = new Directive(REQUIRES_DIRECTIVE)
+            {
+                Arguments = new(new QueryArgument<NonNullGraphType<FieldSetGraphType>>() { Name = FIELDS_ARGUMENT }),
+            };
+            requiresDirective.Locations.Add(DirectiveLocation.FieldDefinition);
+            schema.Directives.Register(requiresDirective);
+        }
+    }
+
+    /// <summary>
+    /// Applies a <c>@link</c> directive to the specified GraphQL schema based on the provided <see cref="FederationDirectiveEnum"/> values.
+    /// </summary>
+    /// <param name="schema">The GraphQL schema to build the link extension for.</param>
+    /// <param name="import">The <see cref="FederationDirectiveEnum"/> values specifying which directives to include in the link extension.</param>
+    public static void ApplyLinkDirective(this ISchema schema, FederationDirectiveEnum import)
+    {
+        schema.ApplyDirective("link", d =>
+        {
+            d.AddArgument(new DirectiveArgument(URL_ARGUMENT) { Value = "https://specs.apollo.dev/federation/v2.0" });
```

Comment:
Is there a plan to support higher [specs](https://www.apollographql.com/docs/federation/federation-versions/) versions?
From what I understand, higher specs require clients to upgrade their router version. So, maybe this version can be set dynamically based on the features used. For example, if the `@interfaceObject` directive was applied during the schema definition, the version here should be updated to v2.3.

Reply:
I think I'll change the `AddFederation` method argument to be a builder method, so we can add features or configuration settings going forwards.  Perhaps the federation version should be a string property?

Reply:
Yep, I thought about such config as well. It can be a "higher supported spec version" config. Then, we can add a validation that will throw if the user used a feature from a higher version. So, I can see two approaches here:
1. Automatic detection of the minimal required version based on used features
2. Explicit higher version set by configuration

I think both are useful.

Reply:
> Yep, I thought about such config as well. It can be a "higher supported spec version" config. Then, we can add a validation that will throw if the user used a feature from a higher version. So, I can see two approaches here:
> 
> 1. Automatic detection of the minimal required version based on used features
> 2. Explicit higher version set by configuration
> 
> I think both are useful.

Completed